socket.go

```diff
@@ -273,6 +275,12 @@ func (socket *mongoSocket) Release() {
 				recycle = false
 			}
 		}
+		if socket.expiryTime != nil && time.Now().After(*socket.expiryTime) {
```

Comment:
I didn't catch this on the first pass. Do we not want to check for expiry when *acquiring* the connection, rather than releasing it? Otherwise the application might still use horrendously old connections *one*, before finally dropping them. If you have an idle server not doing a lot of requests, you could envision something like changing DNS and believing traffic was all shed - except there are stragglers that are violating the expiry time by exactly one request that may show up later.